server.js

```diff
@@ -1,269 +1,354 @@
         }
+        reply(sub)
       }
     },
-    handler: function (req, reply) {
-      var sub = subscriptions.get(req.params.id)
-      var pos = req.query.pos || '0'
-      var num = req.query.num || 1000
-      if (sub) {
-        reply(db.read(pos, num, sub.filter))
-      }
-      else {
+    {
+      method: 'DELETE',
+      path: '/v0/subscription/{id}',
+      config: {
+        auth: {
+          strategy: 'fxa-oauth',
+          scope: ['notify']
+        }
+      },
+      handler: function (req, reply) {
+        subscriptions.remove(req.params.id)
         reply({})
       }
-    }
-  },
-  {
-    method: 'POST',
-    path: '/v0/subscription/{id}/events',
-    config: {
-      auth: {
-        strategy: 'fxa-oauth',
-        scope: ['notify']
+    },
+    {
+      method: 'GET',
+      path: '/v0/subscription/{id}/events',
+      config: {
+        auth: {
+          strategy: 'fxa-oauth',
+          scope: ['notify']
+        },
+        validate: {
+          query: {
+            pos: joi.string().optional(),
+            num: joi.number().min(1).max(1000).optional()
+          }
+        }
       },
-      validate: {
-        payload: {
-          pos: joi.string().required()
+      handler: function (req, reply) {
+        var sub = subscriptions.get(req.params.id)
+        var pos = req.query.pos || '0'
+        var num = req.query.num || 1000
+        if (sub) {
+          db.read(pos, num, sub.filter, reply)
+        }
+        else {
+          reply({})
         }
       }
     },
-    handler: function (req, reply) {
-      var sub = subscriptions.get(req.params.id)
-      if (sub) {
-        sub.pos = req.payload.pos
-        reply(db.read(sub.pos, 1000, sub.filter))
-      }
-      else {
-        reply({})
+    {
+      method: 'POST',
+      path: '/v0/subscription/{id}/events',
+      config: {
+        auth: {
+          strategy: 'fxa-oauth',
+          scope: ['notify']
+        },
+        validate: {
+          payload: {
+            pos: joi.string().required()
+          }
+        }
+      },
+      handler: function (req, reply) {
+        var sub = subscriptions.get(req.params.id)
+        if (sub) {
+          sub.pos = req.payload.pos
+          db.read(sub.pos, 1000, sub.filter, reply)
+        }
+        else {
+          reply({})
+        }
       }
     }
-  }
-])
+  ])
+}
+
+Server.prototype.start = function () {
+  return this.db.start()
+    .then(
+      function () {
+        var d = P.defer()
+        this.hapi.start(
+          function () {
+            log.info('start', this.config.server)
+            d.resolve(this)
+          }.bind(this)
+        )
+        return d.promise
+      }.bind(this)
+    )
+}
+
+Server.prototype.stop = function () {
+  var d = P.defer()
+  this.hapi.stop(
+    function () {
+      log.info('stopped')
+      d.resolve()
+    }
+  )
+  return d.promise
+    .then(
+      function () {
+        return this.db.stop()
+      }.bind(this)
```

Comment:
By the way, I'm interested in your proposal to use es6 features in this repo, and let us avoid all this `bind(this)` boilerplate :-)